modules/video/misc/java/test/TrackerCreateTest.java

```diff
@@ -1,31 +1,84 @@
 package org.opencv.test.video;
 
+import java.io.File;
 import org.opencv.core.Core;
 import org.opencv.core.CvType;
 import org.opencv.core.CvException;
 import org.opencv.core.Mat;
 import org.opencv.core.Rect;
+import org.opencv.dnn.Dnn;
+import org.opencv.dnn.Net;
 import org.opencv.test.OpenCVTestCase;
 
 import org.opencv.video.Tracker;
 import org.opencv.video.TrackerGOTURN;
+import org.opencv.video.TrackerGOTURN_Params;
+import org.opencv.video.TrackerNano;
+import org.opencv.video.TrackerNano_Params;
+import org.opencv.video.TrackerVit;
+import org.opencv.video.TrackerVit_Params;
 import org.opencv.video.TrackerMIL;
 
 public class TrackerCreateTest extends OpenCVTestCase {
 
+    private final static String ENV_OPENCV_DNN_TEST_DATA_PATH = "OPENCV_DNN_TEST_DATA_PATH";
+    private final static String ENV_OPENCV_TEST_DATA_PATH = "OPENCV_TEST_DATA_PATH";
+    private String testDataPath;
+
     @Override
     protected void setUp() throws Exception {
         super.setUp();
-    }
 
+        // relys on https://developer.android.com/reference/java/lang/System
+        isTestCaseEnabled = System.getProperties().getProperty("java.vm.name") != "Dalvik";
+
+        if (isTestCaseEnabled) {
+            testDataPath = System.getenv(ENV_OPENCV_DNN_TEST_DATA_PATH);
+            if (testDataPath == null)
+                testDataPath = System.getenv(ENV_OPENCV_TEST_DATA_PATH);
+            if (testDataPath == null)
+                throw new Exception(ENV_OPENCV_TEST_DATA_PATH + " has to be defined!");
+        }
+    }
 
     public void testCreateTrackerGOTURN() {
+        Net net;
         try {
-            Tracker tracker = TrackerGOTURN.create();
-            assert(tracker != null);
+            String protoFile = new File(testDataPath, "dnn/gsoc2016-goturn/goturn.prototxt").toString();
+            String weightsFile = new File(testDataPath, "dnn/gsoc2016-goturn/goturn.caffemodel").toString();
+            net = Dnn.readNetFromCaffe(protoFile, weightsFile);
```

Comment:
GOTURN files can be located in separate folders - weights in OPENCV_DNN_TEST_DATA_PATH, prototxt in OPENCV_TEST_DATA_PATH.

Reply:
I fixed data search logic and introduced `modelsDataPath` to find files correctly.